pandora-common/src/validation.ts

```diff
@@ -37,6 +37,19 @@ export function ZodArrayWithInvalidDrop<ZodShape extends ZodTypeAny, ZodPreCheck
 	});
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface ZodOverridableType<Output = any, Def extends ZodTypeDef = ZodTypeDef, Input = Output> extends ZodEffects<ZodType<Output, Def, Input>, Output, Input> {
+	override: ((attachedValidation: ((arg: Output, ctx: RefinementCtx) => void)) => void);
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function ZodOverridable<Output = any, Def extends ZodTypeDef = ZodTypeDef, Input = Output>(schema: ZodType<Output, Def, Input>): ZodOverridableType<Output, Def, Input> {
+	let attachedValidation: ((arg: Output, ctx: RefinementCtx) => void) | undefined;
+	const refined = schema.superRefine((arg, ctx) => attachedValidation?.(arg, ctx)) as ZodOverridableType<Output, Def, Input>;
+	refined.override = (fn) => attachedValidation = fn;
```

Comment:
I like this solution much more than the previous one, but I would avoid adding methods to the existing interface. This has high potential to break whenever Zod updates in ways we won't predict.
I think creating a wrapper would be better. Alternatively not having the override on the type itself, but returning a tuple from this function.